detox/local-cli/test.js

```diff
@@ -240,16 +240,22 @@ module.exports.handler = async function test(program) {
   }
 
   function runJest() {
-    if (platform === 'android' && program.workers !== 1) {
-      log.warn('Can not use -w, --workers. Parallel test execution is only supported on iOS currently');
-      program.w = program.workers = 1;
+    const hasMultipleWorkers = ((program.workers || 0) > 1);
+    if (platform === 'android') {
+      program.readOnlyEmu = false;
+      if (hasMultipleWorkers) {
+        program.readOnlyEmu = true;
+        log.warn('Multiple workers is an experimental feature on Android and requires an emulator binary of version 28.0.1 or higher. ' +
+          'Check your version by running: $ANDROID_HOME/tools/bin/sdkmanager --list');
+      }
     }
+    program.reportWorkerAssign = hasMultipleWorkers;
```

Comment:
There is my gut feeling that something is wrong with making semi-anonymous `program` parameters which get handled elsewhere by very optional Jest-specific reporters. Looks implicit, distributed and not really unmanageable to me. 😕 

Are you sure this is the only way?

Reply:
Well the alternative is to just propagate a `multipleWorkers=true/false` an push the operative logic down to actual consumers (reporters in this case). I believe the logic should be separated. Nevertheless I agree we're on the verge of config-vars boom. Perhaps in this case it's safe to always report assignment (unconditionally), in case the reporter was registered.

In any case I admit contemplating the two approaches beforehand, couldn't conclude an absolute case to go with either.

Reply:
> Perhaps in this case it's safe to always report assignment (unconditionally), in case the reporter was registered, since the output is minimal.

Agree.